addons/udes_stock_picking_batch/models/stock_picking.py

```diff
@@ -1,5 +1,55 @@
-from odoo import models
+from odoo import models, api
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
+
+    @api.constrains("batch_id")
+    def _trigger_batch_confirm_and_remove(self):
+        """
+       Batch may require new pickings to be auto confirmed or removed
+       Note: this function is very fragile, any change might brake its functionality
+        """
+        orig_batches = self.env.context.get("orig_batches")
+        if orig_batches:
+            for batch in orig_batches:
+                batch._assign_picks()
+                batch._remove_unready_picks()
+                batch._compute_state()
+        for batch in self.batch_id:
+            batch._assign_picks()
+            batch._remove_unready_picks()
+            batch._compute_state()
+
+    @api.constrains("state")
+    def _trigger_batch_state_recompute(self):
+        """
+        Changes to picking state cause batch state recompute, may also cause
+        unready pickings to be removed from the batch
+        Note: this function is very fragile, any change might brake its functionality
+        """
+        orig_batches = self.env.context.get("orig_batches")
```

Comment:
Can find the batches first and later call the methods , will remove in this way duplicated code

`batches = orig_batches or self.batch_id`
batches._remove_unready_picks() if we can call with multi

Reply:
Addressed